File: sniper-platform/types/strategy.ts
export type Regime = 'TRENDING' | 'MEAN_REVERTING' | 'VOLATILE';

export type Strategy = {
  id: string;
  user_id: string;
  name: string;
  type: string;
  parameters: Record<string, unknown>;
  status: 'active' | 'inactive';
  regime_filters: Regime[];
  created_at: string;
};

File: sniper-platform/apps/sniper-backend/app/models/schemas/strategy.py
from __future__ import annotations

from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field

from app.models.schemas.common import Regime


class StrategyCreate(BaseModel):
    name: str
    type: str
    parameters: dict[str, Any] = Field(default_factory=dict)
    regime_filters: list[Regime] = Field(default_factory=list)


class StrategyUpdate(BaseModel):
    name: str | None = None
    type: str | None = None
    parameters: dict[str, Any] | None = None
    status: str | None = None


class StrategyOut(BaseModel):
    id: str
    user_id: str
    name: str
    type: str
    parameters: dict[str, Any] = Field(default_factory=dict)
    status: str = 'inactive'
    regime_filters: list[Regime] = Field(default_factory=list)
    created_at: datetime
